aiida_quantumespresso/calculations/helpers/__init__.py

```diff
@@ -149,164 +139,166 @@ def pw_input_helper(input_params, structure,
+                        key, input_original_namelists[key], namelist
+                    )
                     if stop_at_first_error:
                         raise QEInputValidationError(err_str)
                     else:
                         errors_list.append(err_str)
-                input_original_namelists[k] = nl
+                input_original_namelists[key] = namelist
 
     # List of the keywords that must not appear in the input
     # (e.g. because they are automatically filled in by the plugin)
-    blocked_kws = [i.lower() for i in
-                   ['pseudo_dir',
-                    'outdir',
-                    'ibrav',
-                    'celldm',
-                    'nat',
-                    'ntyp',
-                    'prefix',
-                    'a', 'b', 'c', 'cosab', 'cosac', 'cosbc',
-                    ]
-                   ]
-    # TODO: possibly add here above also restart_mode?
+    blocked_kws = [
+        i.lower() for i in [
+            'pseudo_dir',
+            'outdir',
+            'ibrav',
+            'celldm',
+            'nat',
+            'ntyp',
+            'prefix',
+            'a',
+            'b',
+            'c',
+            'cosab',
+            'cosac',
+            'cosbc',
+        ]
+    ]
 
     # List of the keywords that must ALWAYS appear in the input
-    compulsory_kws = {i.lower() for i in
-                   ['calculation',
-                    'ecutwfc',
-                     ]
-                   }
+    compulsory_kws = {i.lower() for i in [
+        'calculation',
+        'ecutwfc',
+    ]}
 
     # ===================== PARSING OF THE XML DEFINITION FILE ===============
     module_dir = os.path.dirname(__file__)
     if module_dir == '':
         module_dir = os.curdir
     xml_path = os.path.join(module_dir, 'INPUT_PW-{}.xml'.format(version))
     try:
-        with open(xml_path, 'r') as f:
-            dom = xml.dom.minidom.parse(f)
+        with open(xml_path, 'r') as handle:
+            dom = xml.dom.minidom.parse(handle)
     except IOError:
         prefix = 'INPUT_PW-'
         suffix = '.xml'
-        versions = [fname[len(prefix):-len(suffix)] for fname
-                    in os.listdir(module_dir) if fname.startswith(prefix)
-                    and fname.endswith(suffix)]
-        versions = sorted(versions, key=lambda x: StrictVersion(x))
+        versions = [
+            fname[len(prefix):-len(suffix)]
+            for fname in os.listdir(module_dir)
+            if fname.startswith(prefix) and fname.endswith(suffix)
+        ]
+        versions = sorted(versions, key=lambda x: Version(x))  # pylint: disable=unnecessary-lambda
         strictversions = versions + [version]
-        strictversions = sorted(strictversions, key=lambda x: StrictVersion(x))
+        strictversions = sorted(strictversions, key=lambda x: Version(x))  # pylint: disable=unnecessary-lambda
```

Comment:
I think here pylint might be right: Can't you just put ``key=Version``?

Reply:
Ah yes of course, I forgot that key just takes any callable

Reply:
I fixed it. If you could approve again, I can merge it, thanks!

Reply:
Done! Not sure it helps, the approval needs to be "from someone with write access". But I'm guessing you have the power to merge anyway?